File: codeeval/js/capitalize_words.js
/*
 *
 * Capitalize Words - Code Eval Challenge
 * Tyler Childs - 12/10/2013
 *
 * */

var cap = function(s){
	for(var i=0;i < s.length; i++){
		letter = s[i][0].toUpperCase();

		s[i] = letter + s[i].substring(1);
	}

	return s;
};

var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        
        words = line.split(' ');

        s = cap(words);

        console.log(words.join(' '));
    }
});

File: codeeval/ruby/capitalize_words.rb
#
#
# Capitalize Words - Code Eval Challenge
# Tyler Childs - 10/14/2014
#

class String
  def titleize
    self.split(" ").map do |word|
      word[0].capitalize + word[1..-1]
    end.join(" ")
  end
end

# open file from command line
if ARGV[0].nil?
  puts "NO FILE INPUT"
  exit
end

File.read(ARGV[0]).each_line { |line| puts line.titleize }